psydac/linalg/tests/test_linalg.py

```diff
@@ -618,6 +619,12 @@ def test_inverse_transpose_interaction(n1, n2, p1, p2, P1=False, P2=False):
     ### -1,T & T,-1 --- -1,T,T --- -1,T,-1 --- T,-1,-1 --- T,-1,T (the combinations I test)
     ###
 
+    # Square root test
+    scaled_matrix = B * np.random.random() # Ensure the diagonal elements != 1
+    diagonal_values = scaled_matrix.diagonal(sqrt=False).toarray()
+    sqrt_diagonal_values = scaled_matrix.diagonal(sqrt=True).toarray()
+    assert np.allclose(sqrt_diagonal_values, np.sqrt(diagonal_values))
```

Comment:
When using `allclose` in a unit test, it is good practice to specify the relative and absolute tolerances.

In this case I wonder if we could use an equality check, given that ultimately we are calling the same function `numpy.sqrt` in both cases. Please try:
```suggestion
    assert np.array_equal(sqrt_diagonal_values, np.sqrt(diagonal_values))
```

Reply:
Ok, I redid the test with your suggestion, and it works. I'll commit your suggestion.